TestStack.Dossier.Tests/BuildTests.cs

```diff
@@ -48,6 +49,32 @@ public void GivenBuilder_WhenCallingSetExplicitly_ShouldOverrideValues()
         }
 
         [Fact]
+        public void GivenBuilder_WhenCallingSetWithLambda_ShouldInvokeEachTime()
+        {
+            int counter = 2014;
+            var builder = new CustomerBuilder()
+                .Set(x => x.FirstName, "Pi")
+                .Set(x => x.LastName, "Lanningham")
+                .Set(x => x.YearJoined, () => counter++);
```

Comment:
I also wanted to provide an overload that would take a lambda Func from TObj to TValue, so that you could base your value on the partially constructed object (i.e. make sure that AmountPaid was less than AmountOwed, or something),but that would have required too much internal structuring, so I left it for another day.